### PR TITLE
[hotfix] Fix drop database not working in terminal

### PR DESCRIPTION
**Pull request by MarigWeizhi**
## Why are the changes needed?

Close #xxx.

## Brief change log

- Added the `dropNamespace` to support cascade or non-cascade in `SparkUnifiedCatalogBase` for spark 3.3
- spark 3.2 will use false by default

## How was this patch tested?

- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by huyuanfeng2018 — COMMENTED** [review]
Should we add a test to guard against future regression?

**Comment by MarigWeizhi** [comment]
> Should we add a test to guard against future regression?

Added test, PTAL

File: amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java [hunk 1]
```diff
@@ -86,7 +89,20 @@ public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionExcep
   @Override
   public boolean dropNamespace(String[] namespace, boolean cascade)
       throws NoSuchNamespaceException, NonEmptyNamespaceException {
-    return false;
+    String database = namespaceToDatabase(namespace);
+    if (!unifiedCatalog.databaseExists(database)) {
+      throw new NoSuchNamespaceException(namespace);
+    }
+    List<TableIDWithFormat> tables = unifiedCatalog.listTables(database);
+    if (!tables.isEmpty() && !cascade) {
```
Comment (huyuanfeng2018) [inline amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java:97]: Why not drop all tables first and then drop the database?

I think should this logic be implemented directly by the parent (spark-common).
We already have concrete logic in  `SparkUnifiedCatalogBase.dropNamespace`.
so，remove Override is enough.

WDYT?

**Review by huyuanfeng2018 — COMMENTED** [review]
Thanks for the PR!

File: amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java (hunk 1 above)
Comment (zhoujinsong) [reply amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java:97]: I agreed that we may reuse some implementation in the parent class. But I think it is dangerous to delete all tables by default in Spark 3.2 implementation. So wee may:

- Add a method in the base class to support dropping namespace cascade or not.
- Drop namespace by default with false for Spark 3.2

**Comment by MarigWeizhi** [comment]
Thanks for your review @zhoujinsong @huyuanfeng2018   
I updated the code, I implemented the method to support cascade or not in base instead of spark 3.2.

File: amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java [hunk 2]
```diff
@@ -85,8 +85,8 @@ public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionExcep
    */
   @Override
   public boolean dropNamespace(String[] namespace, boolean cascade)
```
Comment (huyuanfeng2018) [inline amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java:87]: Just delete this method, WDYT?

**Review by huyuanfeng2018 — COMMENTED** [review]
just a nit comment

File: amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java (hunk 2 above)
Comment (MarigWeizhi) [reply amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/src/main/java/org/apache/amoro/spark/SparkUnifiedCatalog.java]: deleted

**Review by huyuanfeng2018 — APPROVED** [review]
LGTM.

**Comment by zhoujinsong** [comment]
Thanks for the contribution! @MarigWeizhi 
Thanks for the review! @huyuanfeng2018

**Pull request merged**